workers/data_refinery_workers/processors/smashing_utils.py

```diff
@@ -288,17 +288,44 @@ def get_frame_inputs():
                 job_context["job"].id
             )
 
-    frame_inputs = get_frame_inputs()
-
-    processed_frames = map(process_frame, frame_inputs)
+    def get_chunked_frame_inputs():
+        """Helper method to create generator for chunking get_frame_inputs
+        chunked into generators that return a slice with size
+        MULTIPROCESSING_CHUNK_SIZE"""
+        source = get_frame_inputs()
+        iterator = itertools.islice(source, MULTIPROCESSING_CHUNK_SIZE)
+        while True:
+            item = next(iterator, None)
+            if item is None:
+                return
+            source, iterator = itertools.tee(itertools.chain([item], source))
+            yield itertools.islice(iterator, MULTIPROCESSING_CHUNK_SIZE)
+            # consume the source iterator to keep track of current chunk
```

Comment:
I think we're gonna need a much more comprehensive set of docstrings/comments on this function. As it is it is really hard to tell by reading the code:
  * What we're trying to do, like the general strategy of the chunking/generators/multiprocessing.
  * Why we're trying to do that.
  * What kind of iterable object will be returned and what it's contents are. Are there multiple levels of iterables here? If so what purpose are they serving?
 
I think you tried to describe this in the docstring:
> Helper method to create generator for chunking get_frame_inputs chunked into generators that return a slice with size MULTIPROCESSING_CHUNK_SIZE

But that's rather terse, `get_frame_inputs` is a function (I'm assuming that you mean the output of it), and I think explaining this better might go a long way to making this more understandable.

Reply:
👍 I think this might need another iteration (no pun intended) but let me know if the last commit helps clear things up